modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -0,0 +1,221 @@
+							id={namespace + 'pin-sku'}
+							onChange={(event) => setSku(event.target.value)}
+							placeholder={Liferay.Language.get('type-sku-here')}
+							value={sku}
+						/>
+						<ClayAutocomplete.DropDown
+							active={(!!resource && !!sku) || initialLoading}
+						>
+							<ClayDropDown.ItemList>
+								{(error || (resource && resource.error)) && (
+									<ClayDropDown.Item className="disabled">
+										No Results Found
+									</ClayDropDown.Item>
+								)}
+								{!error &&
+									resource &&
+									resource.results &&
+									resource.results.map((item) => (
+										<ClayAutocomplete.Item
+											key={item.id}
+											match={sku}
+											value={item.name}
+										/>
+									))}
+							</ClayDropDown.ItemList>
+						</ClayAutocomplete.DropDown>
+						{loadingAd && <ClayAutocomplete.LoadingIndicator />}
+					</ClayAutocomplete>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-3 form-group-sm">
+					<label htmlFor={namespace + 'pin-quantity'}>
+						{Liferay.Language.get('quantity')}
+					</label>
+					<ClayInput
+						id={namespace + 'pin-quantity'}
+						onChange={(event) =>
+							setQuantity(parseInt(event.target.value, 10))
+						}
+						type="number"
+						value={quantity}
+					/>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-6 d-flex form-group-sm justify-content-start mt-4">
+					<ClayButton
+						displayType="link"
+						onClick={() => {
+							setRemovePinHandler({
+								handler: true,
+								pin: showTooltip.details.id,
+							});
+							setShowTooltip({
+								details: {
+									cx: null,
+									cy: null,
+									id: null,
+									label: null,
+									linked_to_sku: 'sku',
+									quantity: null,
+									sku: '',
+								},
+								tooltip: false,
+							});
+						}}
+					>
+						{Liferay.Language.get('delete')}
+					</ClayButton>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-6 d-flex form-group-sm justify-content-between mt-4">
+					<ClayButton
+						displayType="secondary"
+						onClick={() =>
+							setShowTooltip({
+								details: {
+									cx: null,
+									cy: null,
+									id: null,
+									label: '',
+									linked_to_sku: 'sku',
+									quantity: null,
+									sku: '',
+								},
+								tooltip: false,
+							})
+						}
+					>
+						{Liferay.Language.get('cancel')}
+					</ClayButton>
+					<ClayButton
+						displayType="primary"
+						onClick={() => {
+							setShowTooltip({
+								details: {
+									cx: showTooltip.details.cx,
+									cy: showTooltip.details.cy,
+									id: showTooltip.details.id,
+									label: pinLabel,
+									linked_to_sku: linkedValue,
+									quantity,
+									sku,
+								},
+								tooltip: false,
+							});
+						}}
+					>
+						{Liferay.Language.get('save')}
+					</ClayButton>
+				</ClayForm.Group>
+			</ClayCard.Body>
+		</ClayCard>
+	);
+};
+
+AdminTooltip.defaultProps = {};
```

Comment:
Is this really needed (non blocking, just curious)?

Reply:
nope